src/services/subscription-deferred-install-service.ts

```diff
@@ -0,0 +1,66 @@
+import { Request } from "express";
+import IORedis from "ioredis";
+import { getRedisInfo } from "config/redis-info";
+import { v4 as newUUID } from "uuid";
+
+export interface SubscriptionDeferredInstallPayload {
+	installationIdPk: number,
+	gitHubServerAppIdPk?: number,
+	gitHubInstallationId: number,
+	orgName: string
+}
+
+const redis = new IORedis(getRedisInfo("SubscriptionDeferredInstallService"));
+
+const REDIS_CLEANUP_TIMEOUT = 2 * 24 * 3600 * 1000;
+
+const toRedisKey = (requestId: string) => "sub_def_inst_serv_req}" + requestId;
+
+const toRequestId = (req: Request) => req.params["requestId"];
+
+export const extractSubscriptionDeferredInstallPayload = async (req: Request) => {
+	try {
+		const requestId = toRequestId(req);
+		if (!requestId) {
+			throw new Error("Empty request ID");
+		}
+
+		const request = await redis.get(toRedisKey(requestId));
+		if (!request) {
+			throw new Error("Unknown request");
+		}
+
+		const parsedPayload = JSON.parse(request) as SubscriptionDeferredInstallPayload;
+		if (!parsedPayload.installationIdPk) {
```

Comment:
I think the first 2 conditional blocks should stay as is but can we convert these 3 checks for parsedPayload to a switch statement just so it's a little more concise and readable? Something like:

```
  switch (true) {
      case !parsedPayload.installationIdPk:
        throw new PayloadExtractionError("No installationIdPk");
      case !parsedPayload.gitHubInstallationId:
        throw new PayloadExtractionError("No gitHubInstallationId");
      case !parsedPayload.orgName:
        throw new PayloadExtractionError("No orgName");
    }
```

Reply:
To be frank, I'd prefer to have explicit ifs. This switch looks like a hack to me... Maybe I'm too old though...

Reply:
I agree to @bgvozdev , on both points.

Reply:
Definitely agree with the latter :wink_: